packages/babel-parser/src/util/whitespace.js

```diff
@@ -23,6 +23,28 @@ export function isNewLine(code: number): boolean {
 
 export const skipWhiteSpace = /(?:\s|\/\/.*|\/\*[^]*?\*\/)*/g;
 
+export const skipWhiteSpaceInLine =
+  /(?:[^\S\n\r\u2028\u2029]|\/\/.*|\/\*.*?\*\/)*/y;
+
+// Skip whitespace and single-line comments, including /* no newline here */.
+// After this RegExp matches, its lastIndex points to a line terminator, or
+// the start of multi-line comment (which is effectively a line terminator),
+// or the end of string.
+export const skipWhiteSpaceToLineBreak = new RegExp(
+  // Unfortunately JS doesn't support Perl's atomic /(?>pattern)/ or
+  // possessive quantifiers, so we use a trick to prevent backtracking
+  // when the look-ahead for line terminator fails.
+  "(?=(" +
+    // Capture the whitespace and comments that should be skipped inside
+    // a look-ahead assertion, and then re-match the group as a unit.
+    skipWhiteSpaceInLine.source +
```

Comment:
Since `skipWhiteSpaceInLine` is only used here, could you inline it similarly to how you did a few lines below?

(ugh, I wish JS supported multiline regexps with comments)

Reply:
I have a use for this variable elsewhere, but it's a bit more intrusive so I didn't include it in this PR to keep it focused.
Do you still prefer inlining it for now, and then perhaps taking it out when I get to a follow-up PR?

Reply:
It's ok as-is then.